fdbserver/include/fdbserver/IEncryptionKeyProvider.actor.h

```diff
@@ -253,22 +255,31 @@ class TenantAwareEncryptionKeyProvider : public IEncryptionKeyProvider {
 			}
 		}
 		if (domainId == SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID) {
-			*domainName = FDB_DEFAULT_ENCRYPT_DOMAIN_NAME;
+			*domainName = FDB_SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_NAME_REF;
 		}
 		return domainId;
 	}
 
-	int64_t getTenant(const KeyRef& key, bool inclusive) {
+	int64_t getTenantId(const KeyRef& key, bool inclusive) {
 		// A valid tenant id is always a valid encrypt domain id.
-		static_assert(ENCRYPT_INVALID_DOMAIN_ID < 0);
-		if (key.size() < TENANT_PREFIX_SIZE || key >= systemKeys.begin) {
+		static_assert(INVALID_ENCRYPT_DOMAIN_ID == -1);
+
+		if (key.size() && key >= systemKeys.begin) {
 			return SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID;
 		}
-		// TODO(yiwu): Use TenantMapEntry::prefixToId() instead.
-		int64_t tenantId = bigEndian64(*reinterpret_cast<const int64_t*>(key.begin()));
-		if (tenantId < 0) {
-			return SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID;
+
+		if (key.size() < TENANT_PREFIX_SIZE) {
+			// Encryption domain information not available, leverage 'default encryption domain'
+			return FDB_DEFAULT_ENCRYPT_DOMAIN_ID;
+		}
+
+		StringRef prefix = key.substr(0, TENANT_PREFIX_SIZE);
+		int64_t tenantId = TenantMapEntry::prefixToId(prefix, EnforceValidTenantId::False);
```

Comment:
In the case that tenants aren't enabled on the cluster this can return something other than -1 in which case we wont use the default encryption domain?

Reply:
We want to use the default encryption domain in that case, I believe

Reply:
I assume it returns value >0 which coincides with a valid tenant? If yes, then it should be handled as part of Tenant-lookup, CP already handles this case and Redwood encryption tenant-lookup path needs to handle it as well (and so would be Backup encryption or any participant encryption code would be. 

TenantEntryMap::prefixToId returns INVALID_TENANT if conversion results in `tenantId < 0`, but it doesn't do TenantMap lookup as expected

Reply:
Then i believe we need to check if the tenant prefix is in the tenant entry map and if not then use the default domain

Reply:
if its the responsibility of the caller of this function to check the tenant map i think we should make it clear at least in a comment

Reply:
I proposed above that we don't enter this block if there are no tenants, which addresses the question about disabled mode. That said, it is handled later (in `getTenantName`) if an ID doesn't correspond with a real tenant, and this can occur even outside disabled mode.

Reply:
`prefixToId` is the mathematical conversion and has no details about existing Tenants being served. It is the responsibility of the caller to check if returned ID corresponds to a valid tenant.

Reply:
> tenant map i think we should make it clear at least in a comment

I don't think so, `TenantMapEntry` is in-memory representation of Tenant-information, the same struct provides `prefixToId`, hence, it doesn't maintain any Tenant mapping for the cluster. The participant process such as: CP, SS construct in-memory tenant mapping based on the observed transactions or mutations served by using `TenantMapEntry` as in-memory representation

Reply:
Then could we add a comment to this function stating that the tenant id returned may not always be valid even if it is > 0?